src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3213,6 +3213,58 @@ private BoundExpression BindArgumentExpression(BindingDiagnosticBag diagnostics,
         }
 
 #nullable enable
+        private void CheckRefArguments<TMember>(
+            MemberResolutionResult<TMember> methodResult,
+            AnalyzedArguments analyzedArguments,
+            BindingDiagnosticBag diagnostics)
+            where TMember : Symbol
+        {
+            if (analyzedArguments.HasErrors)
+            {
+                return;
+            }
+
+            var result = methodResult.Result;
+            var parameters = methodResult.LeastOverriddenMember.GetParameters();
+
+            for (int arg = 0; arg < analyzedArguments.Arguments.Count; arg++)
+            {
+                if (arg >= parameters.Length)
+                {
+                    // We can run out of parameters before arguments. For example: `M(__arglist(x))`.
```

Comment:
>// We can run out of parameters before arguments. For example: `M(__arglist(x))`. [](http://example.com/codeflow?start=20&length=81)

Perhaps we should check for that exact situation directly, i.e. skip ```BoundArgListOperator``` as an argument.